src/amount.rs

```diff
@@ -35,6 +36,26 @@ impl FromStr for Amount {
     }
 }
 
+impl TryInto<PaymentAmount> for Amount {
+    type Error = AmountParseError;
+
+    fn try_into(self) -> Result<PaymentAmount, Self::Error> {
+        // Convert the payment amount appropriately
+        let minor_units: i64 = self
+            .try_into_minor_units()
+            .ok_or(AmountParseError::InvalidValue)?;
+
+        // Squash into PaymentAmount
+        let payment_amount = if minor_units < 0 {
+            PaymentAmount::pay_customer(minor_units.abs() as u64)
+        } else {
+            PaymentAmount::pay_merchant(minor_units as u64)
+        }?;
+
+        Ok(payment_amount)
```

Comment:
Non-blocking: Since you're using implicit return, you can skip the assignment:
```rust
Ok(if minor_units < 0 {
    PaymentAmount::pay_customer(...)
} else {
    PaymentAmount::pay_merchant(...)
})
```

Reply:
Thank you, fixed.